### ISSUE TITLE
Push typing messages in chat component.

### ISSUE THREAD

**Issue by vibhanshuc** [sevenleaps/chat-template#17]
I am using chat component and trying to push a typing message for the duration between enter key press and reply comes back. This is the modified addMessage function:

``` javascript
addMessage(value) {
    const message = {
      message: value,
      inbound: false,
      backColor: '#dcf8c6',
      duration: 800,
    };
    this.state.messages.push({type: 'typing',
      duration: 500,
      inbound: true});
    this.setState({messages: this.state.messages});
    setTimeout(() => {
      this.state.messages.push(message);
      this.setState({ messages: this.state.messages });
    }, 500);
  }
```

This works fine except it pushes the typed message twice. Can you please help me debug this?

Chat-Template: 0.0.24
React: 15.4.2
Chrome: 56.0.2924.87

**Comment by paulofla** [comment]
Hi @vibhanshuc 

Thanks for opening an issue and providing technical details and a code example!

I'll look into it now and get back to you as soon as I can!

Thanks,
Paul

**Comment by paulofla** [comment]
HI @vibhanshuc 

**---- Explanation of what happened ----**
The `messagesToBeDisplayed` object has the typing and the message when you set the state.

A typing message is treated differently to a normal message and is not added to the internal messages queue. Instead it updates `Conversation` component's `isTyping` state. This shows the typing imagery until another message arrives and/or duration is up.

As the typing message wasn't added to the messages queue, the messages queue will show a smaller length as compared to the next incoming props. (In `componentWillReceiveProps` in the `Conversation` component.)

When you add a message next time it thinks it missed displaying the last message. It will take the last message and add it to the queue to display.

**---- Fix you can implement in your code ----**
In the meantime, here is what you should change it to. (I filter out the typing messages that are in the state and then add a typing message and then the message. I've removed the timeout you've added as the typing message has a duration.)
``` javascript
addMessage(value) {
    const message = {
      message: value,
      inbound: false,
      backColor: '#dcf8c6',
      duration: 800,
    };

    let messages = this.state.messages;
    messages = messages.filter((element) => element.type !== 'typing');
    messages.push({type: 'typing', duration: 500, inbound: true});
    messages.push(message);
    this.setState({ messages: messages });
  }
```

**---- In the future ----**
I'll fix this bug in the next release, which I'm hoping should be today or tomorrow.

Thanks,
Paul

**Comment by paulofla** [comment]
This should be fixed in the 0.0.26 release.
